### PR TITLE
Development

### PR DESCRIPTION
**Pull request by CodeByDani**
check it and tell me if anythings wrong

### CODE REVIEW

File: controller/sendsms_handler.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	"github.com/cyneptic/letsgo-smspanel/internal/core/ports"
+	"github.com/cyneptic/letsgo-smspanel/internal/core/service"
+	"github.com/labstack/echo/v4"
+)
+
+type SendSMSHandler struct {
+	svc ports.SendSMSServiceContract
+}
+
+func NewSnedSMSHandler() *SendSMSHandler {
```
Comment (cyneptic) [inline controller/sendsms_handler.go]: I think there's a typo here, Send is written as Sned.

File: controller/sendsms_handler.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		svc: svc,
+	}
+}
+
+func AddSendSMSRouters(e *echo.Echo) {
+	handler := NewSnedSMSHandler()
+	e.POST("/sendcontactlist", handler.SendToContactListHandler)
+	e.POST("/senduser", handler.SendToUserHandler)
+	e.POST("/sendnumber", handler.SendToNumberHandler)
+	e.POST("/sendcontactlistinterval", handler.SendToContactListIntervalHandler)
```
Comment (cyneptic) [inline controller/sendsms_handler.go]: its hard to read lowercase addresses when they have multiple words, I suggest using hyphens (-) to put some space between the words.
i.e:
send-message-list
send-message-user
send-message-number
send-message-interval

File: controller/sendsms_handler.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	}
+	messageSample, err := validators.ValidatorReciveMessage(jsonMessage)
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	err = h.svc.SendToContactList(messageSample)
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	return c.String(http.StatusOK, "Fine")
```
Comment (cyneptic) [inline controller/sendsms_handler.go]: I don't think "fine" is a good response to a user requesting a message.
I suggest maybe returning a more suitable message or nothing at all, http.StatusOK should be enough.

File: controller/sendsms_handler.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	}
+	messageSample, err := validators.ValidatorReciveMessage(jsonMessage)
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	err = h.svc.SendToUser(messageSample)
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	return c.String(http.StatusOK, "Fine")
```
Comment (cyneptic) [inline controller/sendsms_handler.go]: I don't think "fine" is a good response to a user requesting a message.
I suggest maybe returning a more suitable message or nothing at all, http.StatusOK should be enough.

File: controller/sendsms_handler.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	}
+	messageSample, err := validators.ValidatorReciveMessage(jsonMessage)
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	err = h.svc.SendToNumber(messageSample)
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	return c.String(http.StatusOK, "Fine")
```
Comment (cyneptic) [inline controller/sendsms_handler.go]: I don't think "fine" is a good response to a user requesting a message.
I suggest maybe returning a more suitable message or nothing at all, http.StatusOK should be enough.

File: controller/sendsms_handler.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+	t, err := validators.ValidateNumber(c.QueryParam("t"))
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+
+	err = h.svc.SendToContactListInterval(messageSample, t)
+	if err != nil {
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	return c.String(http.StatusOK, "Fine")
```
Comment (cyneptic) [inline controller/sendsms_handler.go]: I don't think "fine" is a good response to a user requesting a message.
I suggest maybe returning a more suitable message or nothing at all, http.StatusOK should be enough.

File: controller/validators/sendsms_validator.go [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/cyneptic/letsgo-smspanel/internal/core/entities"
+	"github.com/google/uuid"
+)
+
+func ValidateNumber(p string) (time.Duration, error) {
```
Comment (cyneptic) [inline controller/validators/sendsms_validator.go]: this function conflicts with single responsibility principle and its name is not suitable
a validator function should only validate but its converting values to time.
"ValidateNumber" means its validating a number when in fact its validating a time. and its also making a converstion to time.

I suggest making helpers to help out with the conversion and renaming the function properly.

File: controller/validators/sendsms_validator.go [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+	if err != nil {
+		return 0, errors.New("can't convert str to uint")
+	}
+	// t1 = t1 * 1000000000 * 60 * 60
+	t1 = t1 * uint64(time.Hour)
+	t := time.Duration(t1)
+	return t, nil
+}
+
+func ValidatorReciveMessage(msg entities.MessageReciver) (entities.Message, error) {
```
Comment (cyneptic) [inline controller/validators/sendsms_validator.go]: typo in function name "Receive" is written as "Recive"

File: internal/core/entities/base.go [hunk 9]
```diff
@@ -8,10 +8,10 @@ import (
 )
 
 type DBModel struct {
-	ID         uuid.UUID `gorm:"primaryKey;type:uuid" json:"id"`
+	ID         uuid.UUID `gorm:"primaryKey;type:uuid;Column:id" json:"id"`
```
Comment (cyneptic) [inline internal/core/entities/base.go]: Column is used when the name defined in the struct is different than the name of the column. in this case there is no need for this.

I suggest undoing all unnecessary changes to the files that you are not the author of.

File: internal/core/entities/base.go [hunk 10]
```diff
@@ -8,10 +8,10 @@ import (
...
 	CreatedAt  time.Time
 	ModifiedAt time.Time
-	DeletedAt *gorm.DeletedAt
+	DeletedAt  *gorm.DeletedAt
```
Comment (cyneptic) [inline internal/core/entities/base.go]: I suggest undoing all unnecessary changes to the files that you are not the author of.

File: internal/core/entities/number.go [hunk 11]
```diff
@@ -9,7 +9,7 @@ import (
 type Number struct {
 	DBModel
 	No               string    `gorm:"Column:number" json:"number"`
-	UserId           uuid.UUID `gorm:"type:uuid" json:"user_id"`
+	UserId           uuid.UUID `gorm:"Column:userid;type:uuid" json:"user_id"`
```
Comment (cyneptic) [inline internal/core/entities/number.go]: user_id is the best practice we tend to separate different words with underscore in databases and json.

**Review by cyneptic — CHANGES_REQUESTED** [review]
ContactList is the same as PhoneBook struct.
every user has their own list of phonebooks.
what has happened here is that you are taking all the contacts whenever you are using contactlist.
what we need is to take a specific PhoneBook from a User and send the message to all Contacts of That PhoneBook of that User.

**Comment by cyneptic** [comment]
I need the conflicts resolved in order to review the entire request, please resolve the conflicts.

**Comment by DanialAbedini** [comment]
open branch

**Comment by DanialAbedini** [comment]
open development

**Comment by DanialAbedini** [comment]
open agian

**Comment by DanialAbedini** [comment]
open branch

**Pull request closed without merging**